components/messenger.rst

```diff
@@ -284,17 +284,16 @@ do is to write your own CSV receiver::
 
         public function get(): void
         {
-            $ordersFromCsv = $this->serializer->deserialize(file_get_contents($this->filePath), 'csv');
-
-            foreach ($ordersFromCsv as $orderFromCsv) {
-                $order = new NewOrder($orderFromCsv['id'], $orderFromCsv['account_id'], $orderFromCsv['amount']);
-
-                $envelope = new Envelope($order);
-
-                $handler($envelope);
+            // Receive the envelope according to your transport ($yourEnvelope here),
+            // in most cases, using a connection is the easiest solution.
+            
+            if (null === $yourEnvelope) {
+                return [];
             }
-
-            return [$envelope];
+            
+            // Reject the message if needed
```

Comment:
I don't understand this comment ... should we add another (commented) line of code showing how to reject the message?

Reply:
As this is linked to the transport, I don't know if we can find a generic approach, I've added one if you want to check it 🙂

Reply:
Sorry I can't review because I don't understand this ... but as a reader, the comment was strange because it was like "you can reject the message if you want" OK but here? now? how? is the comment related to the following return? Returning means rejecting? etc.

Reply:
Oh ok, sorry for the wording. 

As `get()` allows you to receive a message and decide if you want to send it to the application, reject it or ack it, the core logic of the `get()` method is to decide the next step. 

As done in the RedisReceiver: 

```php
/**
     * {@inheritdoc}
     */
    public function get(): iterable
    {
        $redisEnvelope = $this->connection->get();
        if (null === $redisEnvelope) {
            return [];
        }
        try {
            $envelope = $this->serializer->decode([
                'body' => $redisEnvelope['body'],
                'headers' => $redisEnvelope['headers'],
            ]);
        } catch (MessageDecodingFailedException $exception) {
            $this->connection->reject($redisEnvelope['id']);
            throw $exception;
        }
        return [$envelope->with(new RedisReceivedStamp($redisEnvelope['id']))];
    }
```

You can say: If there's an error (Doctrine try to catch the DbalException for example), so I need to reject the message (maybe because of its content or others factors).

If the message is rejected, the transport handle the next step.

When we say `return [$yourEnvelope->with(new CustomStamp($yourEnvelope['id']);`, we make the message available for the application 🙂 

Hope it helps 🙂